File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.swift
import Foundation

func solution(_ operations:[String]) -> [Int] {
    var data : [Int] = []
    for operation in operations {
        var op = operation.components(separatedBy: " ")
        if op[0] == "I" {
            data.append(Int(op[1])!)
        } else if operation == "D 1" {
            if let max = data.max() {
                if let index = data.firstIndex(of: max) {
                    data.remove(at: index)
                }
            }
        }
        else if operation == "D -1" {
            if let min = data.min() {
                if let index = data.firstIndex(of: min) {
                    data.remove(at: index)
                }
            }
        }
    }
    if data.count <= 0 {
        return [0,0]
    }
    return [data.max()!, data.min()!]
}

File: 프로그래머스/lv3/42628. 이중우선순위큐/이중우선순위큐.py
def solution(operations):
    data = []
    for op in operations:
        command, num = op.split()
        if command == "I":
            data.append(int(num))
        if op == "D 1":
            try:
                data.remove(max(data))
            except:
                pass
        if op == "D -1":
            try:
                data.remove(min(data))
            except:
                pass
    if data:
        return max(data), min(data)
    return 0, 0